tests/unit/test_lifecycle.py

```diff
@@ -41,6 +47,76 @@ def mock_provider(mocker, mock_instance, fake_provider):
     yield _mock_provider
 
 
+@pytest.fixture()
+def mock_get_instance_name(mocker):
+    with patch(
```

Comment:
is there a specific reason to use `with patch()` here since you're already using `mocker`?

Reply:
Whoops! I switched these to fixtures to `mocker`.